### PR TITLE
Upgrade to Twig2

### PR DESCRIPTION
**Pull request by koxu1996**
Currently I am unable to run tests, but it is appears to be working.  It is BC breaking, because Twig2 requires PHP7.

### CODE REVIEW

**Comment by koxu1996** [comment]
Finally I ran the tests and I see there are few problems to be fixed.

**Comment by koxu1996** [comment]
I am trying to fix last error:
```
1) yiiunit\extensions\twig\ViewRendererTest::testNullsInAr                                            
Twig_Error_Runtime: An exception has been thrown during the rendering of a template ("Object of class yii\db\ActiveQuery could not be converted to string").
```

**Comment by koxu1996** [comment]
I copied most of code from _craftcms_ and managed to fix all problems. I look forward to merge.

**Comment by samdark** [comment]
I've fixed travis build in master. Please merge with your changes. Also, there's no need to remove code coverage.

**Comment by samdark** [comment]
Updated pull request a bit to reflect travis fixes made to master.

**Comment by koxu1996** [comment]
Ok

**Comment by samdark** [comment]
#27

**Comment by koxu1996** [comment]
Are you planing to merge it soon?

**Comment by amarox** [comment]
Yes, but this work is not finished

**Comment by koxu1996** [comment]
So what has to be done?

**Comment by amarox** [comment]
Need to check everything

File: Template.php [hunk 1]
```diff
@@ -8,33 +8,51 @@
...
-                if ($this->env->hasExtension('sandbox')) {
-                    $this->env->getExtension('sandbox')->checkMethodAllowed($object, $item);
-                }
-                return call_user_func_array([$object, $item], $arguments);
-            } else {
-                if ($this->env->hasExtension('sandbox')) {
-                    $this->env->getExtension('sandbox')->checkPropertyAllowed($object, $item);
-                }
-                return $object->$item;
+        if ($object instanceof ElementInterface) {
```
Comment (machour) [inline Template.php]: Reaaaally late to the party here, but where does ElementInterface come from?

@amarox, @koxu1996,  if you have the time, could you possibly have a look at #113?

Comment (koxu1996) [reply Template.php]: @machour I will check this ASAP and let you know.

Comment (koxu1996) [reply Template.php]: @machour I posted answer at the [issue page](https://github.com/yiisoft/yii2-twig/issues/113).

**Pull request merged**